[Zhang Yubin] ip

Kimowarui opened the pull request:
**DukePro**

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes/))

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/Kimowarui/ip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x] Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the main method:

```
public class Launcher {
    public static void main(String[] args) {
        Application.launch(Main.class, args);
    }
}
```

Commit:
Add Gradle support

Kimowarui committed:
Level-1

Kimowarui committed:
Level-2

Kimowarui committed:
Level-3

Kimowarui committed:
Level-4

Kimowarui committed:
Level-5

Kimowarui committed:
Level-6

Kimowarui committed:
A-TextUiTesting

Kimowarui committed:
Some modifications from last week ip

Kimowarui committed:
Auto Save

Kimowarui committed:
Auto Date and Times

justintanyf commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,14 @@
+public class Deadline extends Task {
+
+    protected String by;
```
Should this be more specific as to what it refers to?

justintanyf commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,84 @@
+
+        while(!input.equals("bye")) {
+            String[] parts = input.split(" ");
+            if(input.equals("list")) {
```
Might be cleaner to change to switch statements

justintanyf commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,14 @@
+public class Event extends Task {
+
+    protected String at;
```
Might be good to use a more desceiptive name for the variable

justintanyf commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,22 @@
+    }
+
+    public String getStatusIcon() {
+        return (isDone ? "X" : " "); // mark done task with X
```
Ternary operators might look cleaner when seperated into multiple lines, and there is no need for the brackets.

justintanyf commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,84 @@
+                task.add(event);
+                System.out.println("Now you have " + task.size() + " tasks in your list");
+                System.out.println(line);
+            } else if (parts[0].equals("todo")){
```
I think there should be a space between ")" and "{", so ) {

justintanyf reviewed (commented):
Overall, a really good job!

weichang18 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,84 @@
+                }
+                System.out.println(line);
+            } else if (parts[0].equals("done")) {
+                int index = Integer.valueOf(parts[1]);
+                Task thisTask = task.get(index - 1);
+                thisTask.markDone();
+                System.out.println(line);
+                System.out.println("Nice! I've marked this task as done:\n" + thisTask.toString());
+                System.out.println(line);
```
Maybe for this portion can create an additional class(Example UI) to do all this line of code for you, and just need to call that function from the class so that it looks much neater. Can also be done so on your other if-else statements

weichang18 reviewed (commented):
Overall pretty good,  but it seems like you have not done week 2's task. If you have done so, you should commit it so we can help you review your code.

Kimowarui committed:
Make the program more OOP

Kimowarui committed:
Divide classes into packages

Kimowarui committed:
Add JUnit tests

Kimowarui committed:
Add JUnit tests

Kimowarui committed:
Add JavaDoc comments

Kimowarui committed:
Tweak the code to comply with a coding standard

Kimowarui committed:
Give users a way to find a task by searching for a keyword.

Kimowarui committed:
Merge branch 'branch-A-CodingStandard'

Kimowarui committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/command/Parser.java

Kimowarui committed:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-A-Gradle

Kimowarui committed:
Fix some coding style using Gradle

Kimowarui committed:
Add a GUI to Duke

Kimowarui committed:
Make it more OOP

Kimowarui committed:
Add Assertions feature: Assert input given by user will always be valid

Use Assertions feature (not JUnit assertions) to document important assumptions that should hold at various points in the code.

Add two assert lines, one is in Parser, to ensure that Parser will always parse userInput correctly, and another is on Command, to make sure
the message input can be executed. We do so because it can satisfy our need for safe work in Duke.

Kimowarui committed:
Improve code quality

Kimowarui committed:
Merge branch 'branch-A-CodeQuality'

Kimowarui committed:
Add snooze feature selected from category B, C, or D

Kimowarui committed:
Improve the GUI

Kimowarui committed:
Update README.md

Kimowarui committed:
Add files via upload

Kimowarui committed:
Update README.md

Kimowarui committed:
Set theme jekyll-theme-architect

Kimowarui committed:
Update README.md

Kimowarui committed:
Set theme jekyll-theme-hacker
